extensions/opentelemetry/deployment/src/test/java/io/quarkus/opentelemetry/deployment/instrumentation/GraphQLOpenTelemetryTest.java

```diff
@@ -0,0 +1,335 @@
+        assertEquals("HelloResource.hello", querySpan.getName());
+    }
+
+    @Test
+    void multipleResultQueryTraceTest() {
+        String request = getPayload("query {\n" +
+                "  hellos\n" +
+                "}",
+                null);
+
+        assertSuccessfulRequestContainingMessages(request, "[\"hello, hi, hey\"]");
+
+        spanExporter.assertSpanCount(3);
+
+        final List<SpanData> spans = Collections.unmodifiableList(spanExporter.getFinishedSpanItems(3));
+        assertTimeForSpans(spans);
+        assertHttpSpan(spans);
+
+        final SpanData operationSpan = spans.get(1);
+        assertEquals(spans.get(2).getSpanId(), operationSpan.getParentSpanId());
+        assertGraphQLSpan(operationSpan, "GraphQL", "QUERY", "");
+
+        final SpanData querySpan = spans.get(0);
+        assertEquals(operationSpan.getSpanId(), querySpan.getParentSpanId());
+        assertEquals("HelloResource.hellos", querySpan.getName());
+    }
+
+    @Test
+    void nestedCdiBeanInsideQueryTraceTest() throws ExecutionException, InterruptedException {
+        String request = getPayload("query {\n" +
+                "  helloAfterSecond\n" +
+                "}",
+                null);
+        int iterations = 500;
+        ExecutorService executor = Executors.newFixedThreadPool(50);
+
+        try {
+            CompletableFuture<Void>[] futures = new CompletableFuture[iterations];
+            for (int i = 0; i < iterations; i++) {
+                futures[i] = CompletableFuture.supplyAsync(() -> assertSuccessfulRequestContainingMessages(request, "hello"),
+                        executor);
+            }
+            getNestedCdiBeanTestResult(futures, iterations);
+        } finally {
+            executor.shutdown();
+        }
+    }
+
+    private void getNestedCdiBeanTestResult(CompletableFuture<Void>[] futures, int iterations)
+            throws InterruptedException, ExecutionException {
+        CompletableFuture.allOf(futures).get();
+        int numberOfSpansPerGroup = 4;
+        spanExporter.assertSpanCount(iterations * numberOfSpansPerGroup);
+        final List<SpanData> spans = Collections
+                .unmodifiableList(spanExporter.getFinishedSpanItems(iterations * numberOfSpansPerGroup));
+
+        List<List<SpanData>> spanCollections = spans.stream()
+                .collect(Collectors.groupingBy(
+                        SpanData::getTraceId,
+                        LinkedHashMap::new, // use a LinkedHashMap to preserve insertion order
+                        Collectors.toList()))
+                .values().stream()
+                .collect(Collectors.toList());
+
+        assertEquals(spanCollections.size(), iterations);
+        for (List<SpanData> spanGroup : spanCollections) {
+            assertTimeForSpans(spanGroup);
+            assertHttpSpan(spanGroup);
+
+            final SpanData operationSpan = spanGroup.get(2);
```

Comment:
Using order can be tricky and lead to transient failures.
Using `getSpanByKindAndParentId(...)` is recomended.